pkg/lockless_circular_buffer/circular_buffer.go

```diff
@@ -0,0 +1,330 @@
+package lockless_circular_buffer
+
+import (
+	"runtime"
+	"sync/atomic"
```

Comment:
We standardise on uber atomic. It is nicer to use.

Reply:
Gotcha, will replace.